addons/viewrestriction/XEH_postInitClient.sqf

```diff
@@ -0,0 +1,98 @@
+#include "script_component.hpp"
+
+// Exit on Server and Headless
+if !(hasInterface) exitWith {};
+
+// Exit if third person view is not available
+if !(difficultyEnabled "3rdPersonView") exitWith {};
+
+["SettingsInitialized", {
+    // Exit if module is disabled
+    if (GVAR(mode) == 0) exitWith {};
+
+    // Start proper PFH based on mode (separated from one PFH with many conditions to improve performance)
+
+    // Conditions with cameraView results in visible transitions, ongoing use of switchCamera does not
```

Comment:
Indeed, because cameraView will return true only once already in that view and that happens 50% into the transition. `switchCamera` itself shouldn't have a performance impact though (feel free to profile, I am not experienced with that).